src/modules/formBuilder/components/itemEditor/FormEditorItemProperties.tsx

```diff
@@ -0,0 +1,380 @@
+}) => {
+  // React Hook Form handlers (useForm return object)
+  const {
+    control,
+    getValues,
+    handleSubmit,
+    watch,
+    formState: { isDirty },
+  } = handlers;
+
+  // Item type is not allowed to change in this form
+  const [itemTypeValue] = useState<ItemType>(initialItem.type);
+
+  // Monitor changes to the FormItem.component field
+  const itemComponentValue = watch('component');
+
+  const isAssessment = useMemo(
+    () => (Object.values(AssessmentRole) as [string]).includes(definition.role),
+    [definition.role]
+  );
+
+  const componentOverridePicklist = useMemo(() => {
+    return validComponentsForType(itemTypeValue).map((component) => ({
+      code: component,
+      label: startCase(component.toLowerCase()),
+    }));
+  }, [itemTypeValue]);
+
+  return (
+    <form
+      // handleSubmit validates input before calling onSubmit
+      onSubmit={handleSubmit((item) =>
+        // FIXME: we cast to FormItem because of DeepPartial difference. is this safe?
+        onSave(item as FormItem, initialItem.linkId)
+      )}
+    >
+      <Box sx={{ p: 2 }}>
+        <Stack gap={2}>
+          {itemTypeValue !== ItemType.Group && (
+            <FormEditorItemPreview control={control} />
+          )}
+          <Divider />
+        </Stack>
+
+        <Stack gap={2} sx={{ maxWidth: MAX_INPUT_AND_LABEL_WIDTH, mt: 2 }}>
+          {/* TODO standardize heading typography, use hX */}
+          <Typography>
+            Properties
+            {import.meta.env.MODE === 'development' && (
+              // Temp debug button to log form state
+              <Button
+                size='small'
+                variant='outlined'
+                sx={{ float: 'right', typography: 'body2' }}
+                // eslint-disable-next-line no-console
+                onClick={() => console.log(getValues())}
+              >
+                debug
+              </Button>
+            )}
+          </Typography>
+          {componentOverridePicklist.length > 0 && (
+            <Controller
+              name='component'
+              control={control}
+              // rules={{ required: 'need this' }}
+              render={({ field: { ref, ...field }, fieldState: { error } }) => (
+                <SelectOption
+                  label='Component Override'
+                  options={componentOverridePicklist}
+                  {...field}
+                  textInputProps={{
+                    helperText: error?.message,
+                    error: !!error,
+                    inputRef: ref,
+                  }}
+                />
+              )}
+            />
+          )}
+          <Controller
+            name='linkId'
+            control={control}
+            rules={{
+              required: 'Link ID is required',
+              // TODO: add pattern validation once it is enforced
+              // https://github.com/greenriver/hmis-warehouse/pull/4424
+              // pattern: /^[a-zA-Z_$][a-zA-Z0-9_$]*$/
```

Comment:
@martha you raised a good point about this validation, I turned it off for now, and we can re-enable it once https://github.com/greenriver/hmis-warehouse/pull/4424 is merged.